dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+ * PROJECT:     Recycle bin management
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Enumerates contents of a MS Windows 2000/XP/2003 recyclebin
+ * COPYRIGHT:   Copyright 2006-2007 Hervé Poussineau (hpoussin@reactos.org)
+ *              Copyright 2024 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "recyclebin_private.h"
+#include <shlwapi.h>
+
+class RecycleBin5File : public IRecycleBinFile
+{
+public:
+    RecycleBin5File(
+        IN IRecycleBin5 *prb,
+        IN LPCWSTR Folder,
+        IN PDELETED_FILE_RECORD pDeletedFile,
+        IN OUT LPWSTR pszFullName);
```

Comment:
You could take the opportunity to use the new SAL2 annotations (and in the other recyclebin files you've converted)